### PR TITLE
[DEV][FRONTEND] Issue 56: Sending Invitation, accept or reject an invitation

### PR DESCRIPTION
**Pull request by ynsmrkrkmz**
New invitation modal and api call added
Invitations modal added
Accept and reject functionality

### CODE REVIEW

**Review by Oguzoz1 — DISMISSED** [review]

**Comment by niloaydin** [comment]
@ynsmrkrkmz Can you please change the BACKEND to FRONTEND in the title.

**Comment by niloaydin** [comment]
@ynsmrkrkmz why does you frontend pull request including your backend changes. You already created a PR for the backend? Its going to create a problem during merge

File: communitter-frontend/src/components/InvitationModal.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      setIsLoading(false);
+      console.log(newInvitation);
+    } catch (err) {
+      dispatch(setErrorMessage(err.message));
+      setIsLoading(false);
+    }
+  };
+
+  // Example invitation data
+  const invitationData = [
```
Comment (niloaydin) [inline communitter-frontend/src/components/InvitationModal.js]: why are we using a mockdata?

File: communitter-frontend/src/components/InvitationModal.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+              <tr>
+                <th>Username</th>
+                <th>Role</th>
+                <th>Invited By</th>
+                <th>Date</th>
+                <th>Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {invitationData.map((invitation, index) => (
```
Comment (niloaydin) [inline communitter-frontend/src/components/InvitationModal.js]: we map through a mock data, where did you define the real data, or are you goind to define it?

**Comment by ynsmrkrkmz** [comment]
> @ynsmrkrkmz Can you please change the BACKEND to FRONTEND in the title.

changed

File: communitter-frontend/src/components/InvitationModal.js (hunk 1 above)
Comment (ynsmrkrkmz) [reply communitter-frontend/src/components/InvitationModal.js]: this pr is for sending invitation. when i switch to listing feature, i will change those.

File: communitter-frontend/src/components/InvitationModal.js (hunk 2 above)
Comment (ynsmrkrkmz) [reply communitter-frontend/src/components/InvitationModal.js]: yes i explained in the previous comment

**Comment by ynsmrkrkmz** [comment]
> @ynsmrkrkmz why does you frontend pull request including your backend changes. You already created a PR for the backend? Its going to create a problem during merge

because of that i actually labeled this pr as "waiting for merge" not "ready for review". i created this branch on top of backend.

**Review by niloaydin — DISMISSED** [review]

**Review by niloaydin — DISMISSED** [review]

**Review by Oguzoz1 — DISMISSED** [review]

**Review by niloaydin — DISMISSED** [review]

File: communitter-frontend/src/components/NewInvitationModal.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { Button, Form, Modal } from "react-bootstrap";
+import { useSelector } from "react-redux";
+import { getUserRoleValue, roles } from "../utilities/roles";
+
+const NewInvitationModal = ({ show, onHide, onSubmit, isLoading }) => {
```
Comment (Ozmus) [inline communitter-frontend/src/components/NewInvitationModal.js]: Why do you named it as NewInvitationModal ?

**Review by Ozmus — COMMENTED** [review]
Nice Work

File: communitter-frontend/src/components/NewInvitationModal.js (hunk 3 above)
Comment (ynsmrkrkmz) [reply communitter-frontend/src/components/NewInvitationModal.js]: as it serves to send new invitation and it is a modal.

**Pull request closed without merging**